fs/localfs/shallow_fs.go

```diff
@@ -0,0 +1,118 @@
+package localfs
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/natefinch/atomic"
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/snapshot"
+)
+
+// Helpers to implement storing of "shallow" placeholders for files or
+// directory trees in a restore image. A placeholder for directory d1 is
+// d1.kopia-entry/.kopia-entry. As a result, d1.kopia-entry will stat as
+// a directory and show up nicely in colorized ls output. A placeholder
+// for a file f1 is f1.kopia-entry.
+
+func placeholderPath(path string, et snapshot.EntryType) (string, error) {
+	switch et {
+	case snapshot.EntryTypeFile:
+		return path + SHALLOWENTRYSUFFIX, nil
+	case snapshot.EntryTypeDirectory: // Directories and regular files
+		dirpath := path + SHALLOWENTRYSUFFIX
+		if err := os.MkdirAll(dirpath, os.FileMode(dIRMODE)); err != nil {
```

Comment:
not: weird capitalization of `dIRMODE`

Reply:
also I think all caps is discouraged in golang, I would use `ShallowEntrySuffix` and `dirMode`

Reply:
Fixed in 457117b3

Reply:
done in 457117b3